src/main/java/org/edumips64/ui/swing/DineroFrontend.java

```diff
@@ -137,10 +144,110 @@ public void keyReleased(KeyEvent e) {
       }
     });
 
-    JButton browse = new JButton("Browse...");
+		String[] level = {"1", "2", "3", "4", "5"};
+		String[] type = {"data", "instruction", "unified/mixed"};
+
+		cacheLeveLabel	= new JLabel("Set Cache Level (N)");	//Label for cacheLevel
+		cacheTypeLabel  = new JLabel("Set Cache Type (T)");	//Label for cacheType
+
+		cacheLeveLabel.setPreferredSize(new Dimension(110, 26));
+		cacheLeveLabel.setMaximumSize(new Dimension(120, 26));
+		cacheLeveLabel.setMinimumSize(new Dimension(90, 26));
+
+		cacheTypeLabel.setPreferredSize(new Dimension(110, 26));
+		cacheTypeLabel.setMaximumSize(new Dimension(120, 26));
+		cacheTypeLabel.setMinimumSize(new Dimension(80, 26));
+
+		//combo box defined for cache level
+		cacheLevel = new JComboBox(level);
+		cacheLevel.setPreferredSize(new Dimension(80, 26));
+		cacheLevel.setMaximumSize(new Dimension(100, 26));
+		cacheLevel.setMinimumSize(new Dimension(60, 26));
+		
+		//combo box defined for cache type
+		cacheType = new JComboBox(type);
+		cacheType.setPreferredSize(new Dimension(110, 26));
+		cacheType.setMaximumSize(new Dimension(130, 26));
+		cacheType.setMinimumSize(new Dimension(100, 26));
+
+		//combo box action of cache level - passing combo option to argument for cache panel
+		cacheLevel.addActionListener(new ActionListener(){
+			public void actionPerformed(ActionEvent e){
+				String levelMsg = String.valueOf(cacheLevel.getSelectedItem());
+				argLevel = Integer.parseInt(levelMsg);
+			}
+		});
+		cacheLevel.setSelectedIndex(0);
+
+		//combo box action of cache type - passing combo option to argument for cache panel
+		cacheType.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e){
+				argType = (String.valueOf(cacheType.getSelectedItem())).charAt(0);
+			}
+		});
+		cacheType.setSelectedIndex(2);
+
+		//button for creating cache defined
+		create = new JButton("Create Cache");
```

Comment:
You will need to add `create` to the container `cp`, otherwise it is not visible.

Reply:
Added a few more labels along with `create`. It will be visible along with the new PR.

Reply:
What "new PR"? Do you intend to abandon this PR and submit a new one, or to push more commits to this PR?